src/composables/AttachDialog.js

```diff
@@ -0,0 +1,176 @@
+import {
+  ref,
+  computed,
+} from 'vue';
+import { useDialog } from 'src/composables/Dialog';
+import ReloadUsersEvent from 'src/composables/events/ReloadUsersEvent';
+import ReloadGroupsEvent from 'src/composables/events/ReloadGroupsEvent';
+import ReloadRolesEvent from 'src/composables/events/ReloadRolesEvent';
+import ReloadPermissionsEvent from 'src/composables/events/ReloadPermissionsEvent';
+import SelectEvent from 'src/composables/events/SelectEvent';
+import * as GroupService from 'src/services/GroupService';
+import * as RoleService from 'src/services/RoleService';
+import * as UserService from 'src/services/UserService';
+import { Notify } from 'quasar';
+import { useI18n } from 'vue-i18n';
+import { useUserStore } from 'src/stores/UserStore';
+
+/**
+ * Composable that extends useDialog to manage detach user and access control dialogs.
+ * @param {string} dialogName - Name of the dialog.
+ * @param {string} key - Dialog event key.
+ * @param {string} source - The entity that another entity is being detached from.
+ * @param {string} target - The detached entity.
+ * @param {Function} searchFunction - Function to call.
+ * @returns {object} Object with refs and functions to manage detach access control dialog.
+ */
+export function useAttachDialog(dialogName, key, source, target, searchFunction) {
+  const userStore = useUserStore();
+  const submitting = ref(false);
+  const selectOnly = ref(false);
+  const selected = ref([]);
+  const groupId = ref(null);
+  const roleId = ref(null);
+  const userLogin = ref();
+  const isCurrentUser = computed(() => userLogin.value === userStore.login);
+
+  const { show } = useDialog(key, (event) => {
+    submitting.value = false;
+    groupId.value = event.groupId;
+    roleId.value = event.roleId;
+    userLogin.value = event.userLogin;
+    selected.value = [];
+    selectOnly.value = event.selectOnly || false;
+    if (searchFunction) searchFunction();
+  });
+
+  const { t } = useI18n();
+
+  /**
+   * Dispatch a reload event for users, groups, roles or permissions
+   * depending on the value of the `target` variable.
+   * @returns {void}
+   */
+  function reloadEvent() {
+    if (target === 'user') {
+      return ReloadUsersEvent.next();
+    }
+
+    if (target === 'group') {
+      return ReloadGroupsEvent.next();
+    }
+
+    if (target === 'role') {
+      return ReloadRolesEvent.next();
+    }
+
+    return ReloadPermissionsEvent.next();
+  }
+
+  /**
+   * Invoke the appropriate method from GroupService or RoleService.
+   * @param {string} selectedId - Selected id of the item to be attached.
+   * @returns {Promise<void>} Promise with nothing on success.
+   */
+  async function attach(selectedId) {
```

Comment:
same here rename to associate or invokeAttach

Reply:
done